pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, rtpPath ? "share/kak/autoload/plugins" }:
+rec {
+  buildKakounePlugin = attrs@{
+    name ? "${attrs.pname}-${attrs.version}",
+    namePrefix ? "kakplugin-",
+    src,
+    unpackPhase ? "",
+    configurePhase ? "",
+    buildPhase ? "",
+    preInstall ? "",
+    postInstall ? "",
+    path ? lib.getName name,
+    ...
+  }:
+    stdenv.mkDerivation (attrs // {
+      name = namePrefix + name;
+
+      inherit unpackPhase configurePhase buildPhase preInstall postInstall;
```

Comment:
I don't think these are necessary, since they would be getting inherited from `attrs //`

Reply:
also you may want to do `(lib.removeAttrs [ "namePrefix" ] attrs) //` and remove other values which aren't necessary for the underlying call to mkDerivation.